pkg/controller/podunavailablebudget/pub_controller_test.go

```diff
@@ -754,6 +758,64 @@ func TestPubReconcile(t *testing.T) {
 				return *status
 			},
 		},
+		{
+			name: "test select matched deployment, 10 UnavailablePods(5 ready), 10 DisruptionPods(5 delay) and 5 deletion",
+			getPods: func() []*corev1.Pod {
+				var matchedPods []*corev1.Pod
+				for i := 0; i < 100; i++ {
+					pod := podDemo.DeepCopy()
+					pod.Name = fmt.Sprintf("%s-%d", pod.Name, i)
+					if i >= 20 && i < 25 {
+						pod.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+					}
+					matchedPods = append(matchedPods, pod)
+				}
+				return matchedPods
+			},
+			getDeployment: func() *apps.Deployment {
+				object := deploymentDemo.DeepCopy()
+				object.Spec.Replicas = utilpointer.Int32Ptr(100)
+				return object
+			},
+			getReplicaSet: func() *apps.ReplicaSet {
+				object := replicaSetDemo.DeepCopy()
+				object.Spec.Replicas = utilpointer.Int32Ptr(100)
+				return object
+			},
+			getPub: func() *policyv1alpha1.PodUnavailableBudget {
+				pub := pubDemo.DeepCopy()
+				pub.Annotations[policyv1alpha1.PubProtectTotalReplicas] = "50"
+				for i := 0; i < 10; i++ {
```

Comment:
remove targetRef and set selector before testing?

Reply:
add pod.OwnerReferences = nil before testing.